src/client/Microsoft.Identity.Client/ApiConfig/AcquireTokenForClientParameterBuilder.cs

```diff
@@ -102,6 +103,32 @@ public AcquireTokenForClientParameterBuilder WithMtlsProofOfPossession()
             return this;
         }
 
+        /// <summary>
+        /// Specifies additional cache key components to use when caching and retrieving tokens.
+        /// </summary>
+        /// <param name="cacheKeyComponents">The list of additional cache key components.</param>
+        /// <returns>The builder.</returns>
+        /// <remarks>
+        /// <list type="bullet">
+        /// <item><description>This api can be used to associate certificate key identifiers along with other keys with a particular token.</description></item>
+        /// <item><description>In order for the tokens to be succsesfully retrieved from the cache, all components used to cache the token must be provided.</description></item>
+        /// </list>
+        /// </remarks>
+        [SuppressMessage("Microsoft.ApiDesignGuidelines", "RS0016:Symbol 'WithAdditionalCacheKeyComponents' is not part of the declared public API")]
+        public AcquireTokenForClientParameterBuilder WithAdditionalCacheKeyComponents(IDictionary<string, string> cacheKeyComponents)
```

Comment:
How many times will we have to order this collection? Would it be faster to accept a SortedList or a SortedDictionary instead? 

https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.sorteddictionary-2?view=net-9.0&redirectedfrom=MSDN#remarks

Reply:
I realize there are quite a few casts. I changed it so that we still accept an `IDictionary` but the set is ordered right in the beginning and is consistent throughout the code.